app/api/user.py

```diff
@@ -63,9 +65,23 @@ def update_self_information():
             address=data['address']
         )
 
-        if 'photo' in data['photo']:
-            current_user.set_profile_information(photo=data['photo'])
+    except Exception as exception:
+        return bad_request(exception.args[0])
+    return jsonify(True)
 
+
+@bp.route('/self/update/photo', methods=['POST'])
+@login_required
+def update_photo():
+    try:
+        photo = request.files.get('photo')
+        # TODO: separate function for validation
+
+        photo.seek(0, os.SEEK_END)  # Go to the end of file
+        if photo.tell() / 1024 / 1024 > Constants.MAX_PHOTO_SIZE:
+            raise Exception('Max size of photo is {} MB'.format(Constants.MAX_PHOTO_SIZE))
+
+        current_user.set_profile_information(photo=photo)
     except Exception as exception:
```

Comment:
Too broad exception

Reply:
This fix not for this commit.

Reply:
Create issue or link commit? or any info?

Reply:
Created issue #24